### PR TITLE
many: apparmor prompting meta-PR

### PR DESCRIPTION
**Pull request by olivercalder**
This is the meta-PR to track outstanding changes from the prompting branch. Smaller PRs will be opened to chip away at these differences, with the eventual goal of bringing full apparmor prompting functionality into snapd.

### CODE REVIEW

**Comment by olivercalder** [comment]
Fixed the CLA check by adding a mapping from zygmunt.krynicki@canonical.com to me@zygoon.pl in `.mailmap`. This only applies to new commits, so no previous commits will be affected. After the prompting work is merged, this change could be reverted, in case Zygmunt re-connects zygmunt.krynicki@canonical.com to their GutHub or Launchpad accounts and adds commits to snapd in the future.

**Comment by olivercalder** [comment]
I'm not sure why the CLA check is failing again... all commits under Zygmunt have been mapped to me<at>zygoon<dot>pl, and the only new-ish commit which is authored by zygmunt.krynicki<at>canonical<dot>com is 4d01b69eb810760315cdb25e87a1c1b8be9a3a68 , which already appears in master as well. It would be nice if the CLA checker indicated which commit caused the failure, I may have to adjust it and see if I can figure that out. Commit sign-offs should not be the problem.

**Comment by olivercalder** [comment]
Skipping spread for now so commits to the prompting branch don't waste test runner resources.

File: overlord/ifacestate/apparmorprompting/common/common.go [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+func (constraints *Constraints) RemovePermission(permission string) error {
+	origLen := len(constraints.Permissions)
+	i := 0
+	for i < len(constraints.Permissions) {
+		perm := constraints.Permissions[i]
+		if perm != permission {
+			i++
+			continue
+		}
+		copy(constraints.Permissions[i:], constraints.Permissions[i+1:])
```
Comment (zyga) [inline overlord/ifacestate/apparmorprompting/common/common.go:64]: Perhaps we can use something from the slices package: https://pkg.go.dev/slices#DeleteFunc

**Comment by olivercalder** [comment]
The remaining changes from this PR will be applied in new PRs on top of snapd master. The current state of this branch will be preserved in https://github.com/olivercalder/snapd/tree/prompting-archive

**Pull request closed without merging**